test/orderController.test.js

```diff
@@ -0,0 +1,200 @@
+const { app, connectDB } = require('../src/expressServer'); // Import your Express app here
+const { StatusCodes } = require('http-status-codes');
+const request = require('supertest');
+const { MongoMemoryReplSet } = require('mongodb-memory-server');
+const User = require('../src/models/User');
+const { Order } = require('../src/models/Order');
+const Album = require('../src/models/Album');
+const { loginAndReturnCookie } = require('./test_helper');
+const mongoose = require('mongoose');
+let server;
+let mongooseConnection;
+let mongodb;
+
+jest.mock('stripe', () => {
+  // When `require('stripe')` is called, the below jest mock function will be returned instead
+  return jest.fn(() => ({
+    // When the `require('stripe')(API_KEY)` is called, the below object will be returned instead
+    paymentIntents: {
+      // The `stripe.paymentIntents.create` will be replaced with the below jest mock function that returns our mock data
+      create: jest.fn(() => ({
+        id: 'pi_mock123',
+        client_secret: 'mock_secret',
+      })),
+    },
+  }));
+});
+
+beforeAll(async () => {
+  //A version of the mongo server that supports replica sets must be used for the order controller test cases because Mongo transactions are used
+  mongodb = await MongoMemoryReplSet.create({
+    replSet: { storageEngine: 'wiredTiger' },
+  });
+  const url = mongodb.getUri();
+  process.env.MONGO_URL = url;
+  mongooseConnection = await connectDB(url);
+  server = await app.listen(8001);
+});
+
+afterAll(async () => {
+  await server.close();
+  await mongooseConnection.disconnect();
+  await mongodb.stop();
+});
+
+describe('OrderController API Tests', () => {
+  let user;
+  let album;
+  let userCredentials;
+  let admin;
+
+  beforeAll(async () => {
+    //create a user
+    user = await User.create({
+      email: 'Emily@google.com',
+      password: 'secret',
+      name: 'Emily',
+      username: 'emily123',
+      role: 'user',
+    });
+    // Create an album
+    album = await Album.create({
+      albumName: 'Unique Album',
+      artistName: 'Unique Artist',
+      spotifyUrl: 'https://api.spotify.com/v1/albums/unique',
+    });
+    userCredentials = {
+      email: 'Emily@google.com',
+      password: 'secret',
+    };
+  });
+
+  //create an order -Success case
+  it('should create an order successfully', async () => {
+    const signedCookie = await loginAndReturnCookie(userCredentials);
+
+    //data for order
+    let orderData = {
+      orderItems: [
+        {
+          album: album._id,
+          quantity: 2,
+        },
+      ],
+      subtotal: 100,
+      tax: 0.1,
+      total: 110,
+      user: user._id,
+    };
+
+    const response = await request(app)
+      .post(`/api/v1/orders`)
+      .set('Cookie', signedCookie)
+      .send(orderData);
+
+    expect(response.status).toBe(StatusCodes.CREATED);
+    expect(response.body).toHaveProperty('order');
+    expect(response.body).toHaveProperty('clientSecret');
```

Comment:
Since we did create a mock to handle the Stripe request and told the mock Stripe to return a particular `client_secret`  (on lines 14- 26), we actually have the ability here to make sure that our controller code did the right thing and pases that client secret back to the frontend in the response. So while this is good to check just that the body has the property `clientSecret` , we can go one step further and ensure that the value sent by stripe to the backend, matches the value the backend returns in its response:

```suggestion
    expect(response.body).toHaveProperty('clientSecret');
    expect(response.body.clientSecret).toEqual('mock_secret');
```

Reply:
@akosasante everything seems to be fine, but jest complains about updateOrderStatus function- I guess I am resolving it in a stripe webhook ticket 
![image](https://github.com/Code-the-Dream-School/dd-prac-team4-back/assets/103755090/0e3a66d2-8a2e-4a28-b086-ec334136ef01)